File: DSAlgorithms/Leetcode/c++/arrays_hash/1461-binary_codes_of_size_k.cpp
class Solution {
public:
    bool hasAllCodes(string s, int k) {
        if(k > s.size())
            return false;
       std::unordered_set<std::string> a_set; 
       for(int i = 0; i < (s.size() - k + 1); i++)
            a_set.insert(s.substr(i, k));

       std::cout<<a_set.size()<<" "<<2 << k; 
        return (a_set.size() == (1 << k));
    }
};

File: dsa_algo/leetcode/ts/src/arrays_hash/1461-binary_codes_of_size_k.ts
const hasAllCodes =(s: string, k: number): boolean =>{
    const a_set = new Set<string>();
    if(k > s.length)
        return false;
    for(let i = 0; i < (s.length - k + 1); i++)
        a_set.add(s.substr(i, k));

    return (a_set.size === (1 << k));

};
